src/libostree/ostree-repo.c

```diff
@@ -2659,6 +2659,52 @@ get_remotes_d_dir (OstreeRepo          *self,
     return g_file_resolve_relative_path (sysroot, SYSCONF_REMOTES);
 }
 
+static gboolean
+min_free_space_size_validate_and_convert (OstreeRepo    *self,
+                                          const char    *min_free_space_size_str,
+                                          GError       **error)
+{
+  static GRegex *regex;
+  static gsize regex_initialized;
+  if (g_once_init_enter (&regex_initialized))
+    {
+      regex = g_regex_new ("^([0-9]+)(G|M|T)B$", 0, 0, NULL);
+      g_assert (regex);
+      g_once_init_leave (&regex_initialized, 1);
+    }
+
+  g_autoptr(GMatchInfo) match = NULL;
+  if (!g_regex_match (regex, min_free_space_size_str, 0, &match))
+    return glnx_throw (error, "Failed to match '^[0-9]+[GMT]B$'");
```

Comment:
Nitpick: That error message isn’t going to be very useful to the average user (most people don’t understand regexes), and it would end up bubbling up to gnome-software if someone edited their configuration. How about something like “min-free-space-size should be of the format ‘123TB’, ‘123GB’ or ‘123MB’”?